### DISCUSSION TITLE
Failing to bypass Cloudflare with driver.uc_open_with_reconnect

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2707] [Q&A] [answered]
Hi, 

 I've encountered a recurring issue where my attempts to bypass Cloudflare are only successful roughly 30-50% of the time. Additionally, the CAPTCHA handling process, particularly after clicking the verification checkbox, often fails to verify successfully.
 
 Cloudflare Bypass Attempt: I'm using driver.uc_open_with_reconnect("https://gitlab.com/users/sign_in", 20). However, this method fails to bypass Cloudflare about 50-70% of the time.
 
Turnstile Handling: 

def click_turnstile_and_verify(driver):
    logger.info("Starting click_turnstile_and_verify")
    driver.switch_to_frame("iframe")
    driver.uc_click("span.mark")
    driver.assert_element("img#captcha-success", timeout=3)
    
This method also does not consistently result in a successful bypass.

Is there a typical success rate expected for bypassing Cloudflare?

Are there any recommended adjustments or alternative methods to increase the the success rate? 

Thank you for your help.

**Comment by mdmintz** [comment, marked as answer]
This script is working consistently for me:

```python
from seleniumbase import SB

with SB(uc=True, test=True) as sb:
    url = "https://gitlab.com/users/sign_in"
    sb.driver.uc_open_with_reconnect(url, 3)
    if not sb.is_text_visible("Username", '[for="user_login"]'):
        sb.driver.uc_open_with_reconnect(url, 4)
    sb.assert_text("Username", '[for="user_login"]', timeout=3)
    sb.assert_element('label[for="user_login"]')
    sb.highlight('button:contains("Sign in")')
    sb.highlight('h1:contains("GitLab.com")')
    sb.post_message("SeleniumBase wasn't detected", duration=4)
```

Note that you may need to increase the `reconnect_time` in `uc_open_with_reconnect()` for slower internet connections.

Also, websites will block you if they see too much traffic coming from the same IP Address, so that could be your issue.

Adding `incognito=True` may help in some cases.
